File: Mobile Application/procure_mate/lib/models/purchase_order.dart
import 'package:mongo_dart/mongo_dart.dart';

class PurchaseOrder{
  final ObjectId id;
  final String pOrderId;
  final String itemCode;
  final String itemName;
  final double unitPrice;
  final double qty;
  final String uom;
  final double amount;
  final String date;
  final String dueDate;
  final String supplierId;
  final String siteMngId;
  final String siteId;
  final String location;
  final String status;

  PurchaseOrder(
      this.id,
      this.pOrderId,
      this.itemCode,
      this.itemName,
      this.unitPrice,
      this.qty,
      this.uom,
      this.amount,
      this.date,
      this.dueDate,
      this.supplierId,
      this.siteMngId,
      this.siteId,
      this.location,
      this.status);
}

File: Web Application/backend/models/PurchaseOrder.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

// define a new schema for purchase order
const purchaseOrderSchema = new Schema({
  pOrderId: {
    type: String,
    required: true,
  },
  itemCode: {
    type: String,
    required: true,
  },
  itemName: {
    type: String,
    required: true,
  },
  unitPrice: {
    type: Number,
    required: true,
  },
  qty: {
    type: Number,
    required: true,
  },
  uom: {
    type: String,
    required: false,
  },
  amount: {
    type: Number,
    required: true,
  },
  date: {
    type: Date,
    required: true,
  },
  dueDate: {
    type: Date,
    required: true,
  },
  supplierId: {
    type: String,
    required: true,
  },
  siteMngId: {
    type: String,
    required: true,
  },
  siteId: {
    type: String,
    required: true,
  },
  location: {
    type: String,
    required: true,
  },
  status: {
    type: String,
    required: true,
  },
});

// create a model based on the purchase order schema
const PurchaseOrder = mongoose.model("PurchaseOrder", purchaseOrderSchema);

// export the purchase order model to be used in other parts of the application
module.exports = PurchaseOrder;